pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -27,6 +27,25 @@ let
       })
     generatedDerivations;
 
+  grammarToPlugin = grammar:
+    let
+      name = lib.pipe grammar [
+        lib.getName
+
+        # added in buildGrammar
+        (lib.removeSuffix "-grammar")
+
+        # grammars from tree-sitter.builtGrammars
+        (lib.removePrefix "tree-sitter-")
+        (lib.replaceStrings [ "-" ] [ "_" ])
+      ];
+    in
+
+    runCommand "nvim-treesitter-${name}-grammar" { } ''
+      mkdir -p $out/parser
+      ln -s ${grammar}/parser $out/parser/${name}.so
```

Comment:
the .so will work on linux only. wouldn't something like `${stdenv.hostPlatform.extensions.sharedLibrary}` be better ?

Reply:
as a suggestion 
```suggestion
      ln -s ${grammar}/parser $out/parser/${name}.${stdenv.hostPlatform.extensions.sharedLibrary}
```